src/main/java/soot/toolkits/scalar/ConstantValueToInitializerTransformer.java

```diff
@@ -78,10 +78,10 @@ public void transformClass(SootClass sc) {
     Set<SootField> alreadyInitialized = new HashSet<SootField>();
 
     for (SootField sf : sc.getFields()) {
-      // We can only create an initializer for static final fields that
-      // have a constant value. We ignore non-static final fields as
+      // We can only create an initializer for static fields that
+      // have a constant value. We ignore non-static fields as
       // different constructors might assign different values.
```

Comment:
Isn't this statement invalid now since basically any constructor or method could assign different values?

Reply:
It's fine for a method to assign different values at runtime; in case of non-final fields it is just a default value.
In the previous implementation the initial value just got lost, since no tag was created albeit a default value was specified. Since the transformer puts the initialization at the beginning of the static initializer, it could even be overwritten within the static initializer itself without problems.

Reply:
I see. Thanks for explaining.